java/client/src/org/openqa/selenium/firefox/GeckoDriverService.java

```diff
@@ -48,19 +49,24 @@
    */
   public static final String GECKO_DRIVER_EXE_PROPERTY = "webdriver.gecko.driver";
 
+  private final Duration timeout;
+
   /**
    *
    * @param executable The GeckoDriver executable.
    * @param port Which port to start the GeckoDriver on.
    * @param args The arguments to the launched server.
    * @param environment The environment for the launched server.
+   * @param timeout The timeout for connecting to the browser.
    * @throws IOException If an I/O error occurs.
    */
   public GeckoDriverService(
```

Comment:
Changing the public constructor breaks compatibility. Please add a second, and have the original constructor delegate to that. You may mark the older constructor as `deprecated` so we can then get back to just one in the release after 3.14.